analyses/fusion-frequencies/01-fusion-frequencies.R

```diff
@@ -0,0 +1,242 @@
+  # assert all Kids_First_Biospecimen_IDs are unique
+  stopifnot(identical(nrow(x), length(unique(x$Kids_First_Biospecimen_ID))))
+  
+  fx <- filter(x, !is.na(cancer_group), !is.na(cohort))
+  return(fx)
+})
+
+# Subset tumor samples to filter alteration df ---------------------------
+tumor_kfbids <- htl_df %>%
+  filter(sample_type == 'Tumor') %>%
+  pull(Kids_First_Biospecimen_ID)
+
+
+fusion_df <- fusion_df  %>%
+  # We want to keep track of the gene symbols for each sample-fusion pair
+  dplyr::select(Sample,
+                Kids_First_Participant_ID, 
+                FusionName, 
+                Gene1A, 
+                Gene1B, 
+                Gene2A, 
+                Gene2B, 
+                Fusion_Type,
+                DomainRetainedGene1A,
+                DomainRetainedGene1B,
+                reciprocal_exists,
+                annots,
+                BreakpointLocation,
+                ends_with("anno")) %>%
+  # Update reciprocal_exits to fusion with atleast 1 kinase gene involved
+  mutate(
+         reciprocal_exists_kinase = 
+           # if either gene are kinase we will have "Yes" or "No" values
+           # in DomainRetainedGene1A and DomainRetainedGene1B
+           if_else(is.na(DomainRetainedGene1A) &
+                     is.na(DomainRetainedGene1B) ,
+                   # reciprocl_exists is a logical column
+                   FALSE,
+                   reciprocal_exists)) %>%
+  dplyr::rename("Kids_First_Biospecimen_ID"="Sample",
+         "Kinase_domain_retained_Gene1A" = "DomainRetainedGene1A",
+         "Kinase_domain_retained_Gene1B" = "DomainRetainedGene1B",
+         "Reciprocal_exists_either_gene_kinase" = "reciprocal_exists_kinase")%>%
+  # replace NA to "" in columns that have NA
+  replace_na(list("Kinase_domain_retained_Gene1A"="",
+                  "Kinase_domain_retained_Gene1B"="",
+                  "Reciprocal_exists_either_gene_kinase"="",
+                  "Gene1A_anno"="",
+                  "Gene1B_anno"="",
+                  "Gene2A_anno"="",
+                  "Gene2B_anno"="",
+                  "Fusion_anno"=""))%>%
+  # We want a single column that contains the gene symbols
+  tidyr::gather(Gene1A, Gene1B, Gene2A, Gene2B,
+                key = gene_position, value = gene_symbol) %>%
+  filter(!is.na(gene_symbol)) %>%
+  dplyr::rename(Gene_Position = gene_position,
+                Gene_Symbol = gene_symbol)
+
+print(alt_id)
+
+if (opt$alt_id == "FusionName,Fusion_Type") {
+  fusion_df <- fusion_df %>%
+    mutate( Alt_ID=paste(!!as.name(alt_id),sep="_"))
+  keep_cols <- c("FusionName","Fusion_Type","Gene_Symbol", "Gene_Position",
+                 "Fusion_anno","BreakpointLocation", "annots",
+                 "Kinase_domain_retained_Gene1A",
+                 "Kinase_domain_retained_Gene1B",
+                 "Reciprocal_exists_either_gene_kinase",
+                 "Gene1A_anno","Gene1B_anno","Gene2A_anno","Gene2B_anno")
+  
+} else if (opt$alt_id == "Gene_Symbol") {
```

Comment:
Might be more readable if tested on `alt_id`.

```suggestion
} else if (identical(alt_id, "Gene_Symbol")) {
```